packages/grant-explorer/src/features/common/__tests__/CopyToClipboardButton.test.tsx

```diff
@@ -0,0 +1,43 @@
+import { fireEvent, screen } from "@testing-library/react";
+import { renderWithContext } from "../../../test-utils";
+import CopyToClipboardButton from "../CopyToClipboardButton";
+import { vi } from "vitest";
+
+const textToCopy = "CLICK ME!";
+
+Object.assign(navigator, {
+  clipboard: {
+    writeText: () => {
+      /* do nothing.*/
+    },
+    readText: () => textToCopy,
+  },
+});
+
+describe("<CopyToClipboardButton />", () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it("should display Copy to clipboard when not clicked ", () => {
+    renderWithContext(<CopyToClipboardButton textToCopy={textToCopy} />);
+
+    expect(screen.getByText("Share Profile")).toBeInTheDocument();
+  });
+
+  it("should display Copied to clipboard when clicked", () => {
+    renderWithContext(<CopyToClipboardButton textToCopy={textToCopy} />);
+    const copyButton = screen.getByRole("button");
+    fireEvent.click(copyButton);
+
+    expect(screen.getByText("Link Copied")).toBeInTheDocument();
+  });
+
+  it("should copy value to clipboard when clicked", () => {
+    renderWithContext(<CopyToClipboardButton textToCopy={textToCopy} />);
+    const copyButton = screen.getByRole("button");
+    fireEvent.click(copyButton);
+
+    expect(navigator.clipboard.readText()).toEqual(textToCopy);
```

Comment:
Unless this is a rhetorical question, I believe that it is?

Reply:
I mean, `readText` is defined in the test itself and then asserted in the test itself, doesn't exercise the component at all unless I'm missing something. Is there any modification you could do to the `<CopyToClipboardButton>` that would cause that test to fail?

Something like this might be closer to the spirit of the test:

```tsx
    const clipboard = { writeText: vi.fn() }
    Object.assign(navigator, { clipboard })
    renderWithContext(<CopyToClipboardButton textToCopy={'foobar'} />);
    const copyButton = screen.getByRole("button");
    fireEvent.click(copyButton);
    expect(navigator.clipboard.writeText()).toHaveBeenCalledWith('foobar');
```